### PR TITLE
chore: update accounts related packages

### PR DESCRIPTION
**Pull request by ccharly**
## **Description**

Updating packages to use versions coming from the new [accounts monorepo](https://github.com/MetaMask/accounts).

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/27284?quickstart=1)

## **Related issues**

Related to:

- https://github.com/MetaMask/accounts/pull/39
- https://github.com/MetaMask/accounts/pull/50
- https://github.com/MetaMask/accounts/pull/54
- https://github.com/MetaMask/core/pull/4713
- https://github.com/MetaMask/core/pull/4734
- https://github.com/MetaMask/snap-simple-keyring/pull/156
- https://github.com/MetaMask/snap-watch-only/pull/52
- https://github.com/MetaMask/snap-bitcoin-wallet/pull/255
- https://github.com/MetaMask/snap-account-abstraction-keyring/pull/142

## **Manual testing steps**

Test parts of the extension that closely related to accounts management + HW wallets support.

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -359,7 +359,7 @@
     "@metamask/snaps-rpc-methods": "^11.1.1",
     "@metamask/snaps-sdk": "^6.5.1",
     "@metamask/snaps-utils": "^8.1.1",
-    "@metamask/transaction-controller": "^37.0.0",
+    "@metamask/transaction-controller": "37.1.0",
```
Comment (danroc) [inline package.json:362]: Why did you remove the `^`?

Comment (ccharly) [reply package.json:362]: Good catch, that's a mistake 🙈 thanks!

**Review by danroc — DISMISSED** [review]

File: test/e2e/tests/metrics/state-snapshots/errors-after-init-opt-in-background-state.json [hunk 2]
```diff
@@ -62,16 +62,8 @@
...
-          "0": "string",
-          "1": "string",
-          "2": "string"
-        },
-        "destNetworkAllowlist": {
-          "0": "string",
-          "1": "string",
-          "2": "string"
-        }
+        "srcNetworkAllowlist": { "0": "string", "1": "string", "2": "string" },
```
Comment (ccharly) [inline test/e2e/tests/metrics/state-snapshots/errors-after-init-opt-in-background-state.json:65]: IDK why this got pretty-print differently, maybe our prettier/eslint config got updated. Anyway, the CI is happy with that change :)

**Review by montelaidev — DISMISSED** [review]

**Comment by plasmacorral** [comment]
Tested commit 1fdad0a with Chrome 129 and Firefox 131 in mac sonoma 14.6.1

Tested migration in Chrome from 11.16.16 to 1fdad0a
Tested migration in Firefox from 12.3.0 to 1fdad0a

Was able to add, import pk, remove accounts, add hardware, remove hardware account, forget hardware, add and remove SSK accounts, sign, send, lock and unlock, pin and hide accounts, dapp connect.

**Review by danroc — APPROVED** [review]

**Review by montelaidev — APPROVED** [review]
